api/render/render.go

```diff
@@ -0,0 +1,115 @@
+// Package render implements functionality related to response rendering.
+package render
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/pkg/errors"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/smallstep/certificates/acme"
+	"github.com/smallstep/certificates/api/log"
+	"github.com/smallstep/certificates/authority/admin"
+	"github.com/smallstep/certificates/errs"
+	"github.com/smallstep/certificates/logging"
+	"github.com/smallstep/certificates/scep"
+)
+
+// JSON writes the passed value into the http.ResponseWriter.
+func JSON(w http.ResponseWriter, v interface{}) {
+	JSONStatus(w, v, http.StatusOK)
+}
+
+// JSONStatus writes the given value into the http.ResponseWriter and the
+// given status is written as the status code of the response.
+func JSONStatus(w http.ResponseWriter, v interface{}, status int) {
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(status)
+	if err := json.NewEncoder(w).Encode(v); err != nil {
+		log.Error(w, err)
+
+		return
+	}
+
+	log.EnabledResponse(w, v)
+}
+
+// ProtoJSON writes the passed value into the http.ResponseWriter.
+func ProtoJSON(w http.ResponseWriter, m proto.Message) {
+	ProtoJSONStatus(w, m, http.StatusOK)
+}
+
+// ProtoJSONStatus writes the given value into the http.ResponseWriter and the
+// given status is written as the status code of the response.
+func ProtoJSONStatus(w http.ResponseWriter, m proto.Message, status int) {
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(status)
+
+	b, err := protojson.Marshal(m)
+	if err != nil {
+		log.Error(w, err)
+
+		return
+	}
+
+	if _, err := w.Write(b); err != nil {
+		log.Error(w, err)
+
+		return
+	}
+
+	// log.EnabledResponse(w, v)
+}
+
+// Error encodes the JSON representation of err to w.
+func Error(w http.ResponseWriter, err error) {
+	switch k := err.(type) {
+	case *acme.Error:
+		acme.WriteError(w, k)
+		return
+	case *admin.Error:
+		admin.WriteError(w, k)
+		return
```

Comment:
I'm not entirely sure there's even chance of `acme.Error` or `admin.Error` type of errors to reach this function. I'm trying to work with the dependency graph of that `api` package and if that's indeed the case, this switch statement will be removed completely in tomorrow's PR.

For now, this PR is just moving some code around so that it's easier to understand what's going on.

Reply:
In the first file of the PR you added this, the code will go through that case statement.
```
render.Error(w, acme.NewError(acme.ErrorAccountDoesNotExistType,
		"account does not exist"))
```

Reply:
So, this `case` was there before, but other than that I don't see any other import of `acme` or `admin` in `api`. Do you?

Reply:
There are no other imports, what I'm saying is that if you add the header, you can remove acme.WriteError, and admin.WriteError. It's duplicated code.